sqlite-db/src/failed.rs

```diff
@@ -256,11 +255,12 @@ async fn insert_failed_cfd(
     Ok(())
 }
 
-async fn insert_event_log(
-    conn: &mut Transaction<'_, Sqlite>,
-    id: OrderId,
-    event_log: EventLog,
-) -> Result<()> {
+async fn insert_event_log<'a, A>(conn: A, id: OrderId, event_log: EventLog) -> Result<()>
+where
+    A: Acquire<'a, Database = Sqlite>,
```

Comment:
This refactor is nice! I don't really know exactly how this all works internally, but one thing I would note that in future, for some functions, `Transaction` might be preferable over `impl Acquire`, if the given operation should *always* be part of a bigger transaction, i.e if it could leave the db in an inconsistent state if a subsequent operation fails, if the function is designed only to be a helper function and always a constitutent of a larger operation.

I have no idea if this is ever true in this case, though :sweat_smile:

Reply:
You raise an interesting point. On the one hand, by asking for a `Transaction` you inform the caller that they should be using one and therefore thinking about committing or rolling back. On the other hand, these functions don't need to interface with the `Transaction` API at all: we really don't want to call `commit` inside one of these functions (unless they are the very last thing that needs to be done)!

Given that we don't actually need to (or should) use the functionality provided by the trait inside these functions, I would lean towards passing in something that implements `Execute` (in this case we have to settle for `Acquire`, because of library limitation).